packages/subproviders/src/subproviders/ledger.ts

```diff
@@ -127,21 +128,27 @@ export class LedgerSubprovider extends Subprovider {
     public async getAccountsAsync(): Promise<string[]> {
         this._ledgerClientIfExists = await this.createLedgerClientAsync();
 
-        // TODO: replace with generating addresses without hitting Ledger
+        let ledgerResponse;
+        try {
+            ledgerResponse = await this._ledgerClientIfExists.getAddress_async(
+                `${this._derivationPath}`, this._shouldAlwaysAskForConfirmation, SHOULD_GET_CHAIN_CODE,
+            );
+        } finally {
+            await this.destoryLedgerClientAsync();
+        }
```

Comment:
We should still handle any potential errors here with a `catch` statement.

Reply:
We were not doing that in a previous implementation. How do we want to handle them?
Before we were just re-throwing them. This is the same. Otherwise - we'll need to have a mapping for all possible ledger errors.